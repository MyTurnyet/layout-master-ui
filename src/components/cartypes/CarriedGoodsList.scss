@import "../../theme/theme";

.CarriedGoodsList {
color: $tertiary-light;
  padding: 5px 10px;
  font-size: 1em;
}
