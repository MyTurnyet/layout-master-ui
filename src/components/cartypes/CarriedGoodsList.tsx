import React, {FunctionComponent} from "react";
import styled from "styled-components";

interface CarriedGoodsList {
    goods: string[]
}

const StyledCarriedGoodsList = styled.div(props=>`
  color: ${props.theme.colors.tertiary.light};
  padding: 5px 10px;
  font-size: 1em;`
);

const CarriedGoodsList: FunctionComponent<CarriedGoodsList> = (props) => {
    return (<StyledCarriedGoodsList>
        {props.goods.join(", ")}
    </StyledCarriedGoodsList>);
}

export default CarriedGoodsList;
