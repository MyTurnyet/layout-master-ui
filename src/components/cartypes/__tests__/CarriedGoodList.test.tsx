import React from 'react';
import {render, screen} from '@testing-library/react';
import CarriedGoodsList from "components/cartypes/CarriedGoodsList";

describe(' CarriedGoodList ', () => {
    it('should ', () => {
        const carriedGoodsList: CarriedGoodsList = {
            goods: ["Food", "Logs", "Parts"]
        }
        render(<CarriedGoodsList {...carriedGoodsList}/>);
        const element = screen.getByText("Food, Logs, Parts");
        expect(element).toBeInTheDocument();
        expect(element).toHaveClass("CarriedGoodsList");
    })
});
