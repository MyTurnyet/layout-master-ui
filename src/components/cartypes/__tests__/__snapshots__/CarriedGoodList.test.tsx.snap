// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` CarriedGoodList  should  1`] = `
<div
  class="sc-bdfBwQ fAJcNs"
>
  Food, Logs, Parts
</div>
`;
